### PR TITLE
Yamtar Medical Crate Additions

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
Requested by Leon, thanks TinySaturn and @falcon2346 for sprites!

## What this does

Adds 6 new unique items to the Yamtar trader crate
# 
**Self-refilling autoinjector**
hypo
Works the same as a regular autoinjector (including containing Doctor's Delight, yes, autoinjectors contain Doctor's Delight). However, this interesting medical tool has the ability to gradually refill itself over the course of 5 minutes, allowing it to be reused for the next emergency.

**Advanced emergency defibrillator**
defib
Through the power of SCIENCE, it works on targets that are wearing clothes at a 100% chance, and slowly regains a charge every 20 seconds by draining the local APC's power remotely. In addition, it's also slightly faster. Doctors will never be forced to take off the engineer's hardsuit with this tool!

**A virus dish of an Anomalous Meme**
dish
Contains a sample of an Anomalous Meme, a variant of a Meme disease that can carry all 4 stages rather than just 2, and develops twice as fast. A delight to virologists everywhere, both good and bad.

**A pill bottle of Panacea**
canister pill
Contains 3 doses (3 pills containing 0.2u) of an incredibly potent and very limited reagent that cures any injury, disease, missing limbs and even death. Like nectar stolen from the gods, this too was stolen from the space gods.

**Antibody resetter**
resetter

A powerful tool for Virologists that wish to make use of their viruses after a cure has already been widely distributed, this spiky tool resets an user's antibodies from anywhere between 1% to 5%, ready to host a new disease.

**Advanced surgeon tools**
box
A box containing a set of advanced surgery tools, the same as the ones created in R&D. A great addition in any surgery room, especially when R&D is not staffed.

#
In addition, it is slightly less likely to get a suit sensing wand and the crate comes with 3 freebies, each of which can be either a medbot cube to help Medbay patients or an advanced first-aid kit.

## Why it's good
A nice set of tools for Medical to enjoy, generally attempting to address certain grievances that might arise when doing Medical work, and their usefulness will make them greatly coveted for traders to sell off and gain a profit, making it about as enticing as the Alcatraz crate.

## How it was tested
Spawning in Medbay, using the self-refilling autoinjector, using the defibrillator on a patient, using the defibrillator on a patient wearing a hardsuit, using an emagged defibrillator on a patient (it works the same), infecting with the virus dish, using the antibody resetter on a patient and seeing the difference in antigens, consuming Panacea after taking damage, using Panacea on a dead patient, opening the advanced surgeon tools box.

## Changelog
:cl:
 * rscadd: Added a wider amount of content to the Trader's "Yamtar" crate, containing unique items that can greatly aid in medical endeavors.
 * tweak: The "Yamtar" crate is slightly less likely to have a suit sensing wand.

### CODE REVIEW

**Comment by SonixApache** [comment]
\>`rejuvenate()` juice
no

**Comment by boy2mantwicethefam** [comment]
It's in an extremely limited supply but I can certainly lower the number of pills, all the way down to 1, if necessary. Death is already plenty cheap to come back from for the overwhelming majority of casualties, even changeling victims can be cloned nowadays, and my point of view is from the perspective of how long it could take to bring a patient back to life. It's a ticket to turn a 3-minute patient into a 2-second one, and it carries the risk of reviving a patient that can't come back to the round compared to the conventional revival methods. It also doesn't work if it's not at least 0.2u when consumed (and it has to be ingested, so no "skin contact" gas clouds), so any shenanigans that come from chemical gas clouds and cryo chem multiplication should be ineffective. Hopefully.

If it proves particularly contentious and disliked by people I can always remove it.

**Comment by GreatBigFailure** [comment]
The panacea SCP is one of my favorites. Requesting a small flavor addition that would be far more work than it would be worth:
Add snowflake interaction to either mechanic scanners or the xenoarch artifact reader thing. Always destroys the panacea pill and has a 1% chance of allowing more to be fabricated/recreated in some way. 

Also good additions. I think you got the balance just right for trader items. Desirable but not round defining.

File: code/modules/reagents/reagents/reagents_medical.dm [hunk 1]
```diff
@@ -82,6 +82,25 @@
...
 	T.add_planthealth(50)
 
+/datum/reagent/panacea
+	name = "Panacea"
+	id = PANACEA
+	description = "A variant of Adminordrazine that has been subjected to medical sciences to make it incredibly potent. It's magic, stolen from the gods."
+	reagent_state = REAGENT_STATE_LIQUID
+	color = "#EECE19" //rgb: 238, 206, 25
+	density = ARBITRARILY_LARGE_NUMBER
+	specheatcap = ARBITRARILY_LARGE_NUMBER
```
Comment (Eneocho) [inline code/modules/reagents/reagents/reagents_medical.dm:92]: These two unironically make it more attractive as an engineering chem than a medicine, the infinite TEG.

**Comment by Eneocho** [comment]
image

**Comment by Eneocho** [comment]
I don't see what sonix's issue with the panacea pill is, it's on limited supply and is basically just better hivecores, which can be acquired legitimately and in a pseudo-infinite amount (as long as there's money to buy more advanced lazarus injectors).

**Comment by boy2mantwicethefam** [comment]
@GreatBigFailure I want to keep things mostly as it is for now and not deviate from what people vote for unless it's something they hate and enough people want removed. I'm not sure I like the idea of a tiny chance to let it be scannable because a Mechanic would have an incentive to hoard the life-giving pills and then burn through them for a tiny chance to fix everything for the rest of the round, and for now I want to leave the Panacea as is (aside from getting sprites).

**Comment by Blithering** [comment]
I kinda want a chance for the panacea to be replaced with panna cotta by those sneaky traders.

**Pull request merged**